File: ILTISCoreProject/tests/ch.hsr.ifs.iltis.core.core.tests/src/ch/hsr/ifs/iltis/core/core/tests/data/WrapperTest.java
package ch.hsr.ifs.iltis.core.core.tests.data;

import static org.junit.Assert.assertEquals;

import java.util.Arrays;

import org.junit.Test;

import ch.hsr.ifs.iltis.core.core.data.Wrapper;


public class WrapperTest {

   @Test
   public void lambdaEscape() {
      final Wrapper<String> w = new Wrapper<>("");

      final String[] strings = { "foo", "bar", "baz", "foobar", "foobaz" };
      Arrays.stream(strings).forEach(s -> {
         if (s.contains("foo")) {
            w.wrapped += s;
         }
      });

      assertEquals("foofoobarfoobaz", w.wrapped);
   }

}

File: ILTISCoreProject/tests/ch.hsr.ifs.iltis.core.core.tests/src/ch/hsr/ifs/iltis/core/core/tests/data/WrapperTest.kt
package ch.hsr.ifs.iltis.core.core.tests.data;

import org.junit.Assert.assertEquals;

import java.util.Arrays;

import org.junit.Test;

import ch.hsr.ifs.iltis.core.core.data.Wrapper;


class WrapperTest {

	@Test
	fun lambdaEscape() {
		val w = Wrapper("")

		val strings = arrayOf("foo", "bar", "baz", "foobar", "foobaz")
		Arrays.stream(strings).forEach {
			if (it.contains("foo")) {
				w.wrapped += it;
			}
		}

		assertEquals("foofoobarfoobaz", w.wrapped);
	}

}
